composed-message-processor/composed-message-processor.bal

```diff
@@ -0,0 +1,66 @@
+import ballerina/http;
+
+type InfoRequest record {|
+    string[] states;
+|};
+
+type StateInfo record {|
+    decimal revenue;
+    decimal operatingExpenses;
+    int production;
+    int totalEmployees;
+|};
+
+type AggregratedInfo record {|
+    map<decimal> revenueByState;
+    decimal totalRevenue;
+    decimal maxRevenue;
+    string maxRevenueState;
+    map<decimal> operatingExpensesByState;
+    decimal totalOperatingExpenses;
+    int totalProduction;
+    map<int> productivityByState;
+|};
+
+final map<string> stateRoutes = {
+    Texas: "http://api.texas.office.com.balmock.io",
+    Ohio: "http://api.ohio.office.com.balmock.io",
+    Florida: "http://api.florida.office.com.balmock.io"
+};
+
+AggregratedInfo summary = {
```

Comment:
defining this here is wrong.

Reply:
I guess we need have a function named createEmptySummary. This function will create an return the summary record. Use the created record value in the resource method to do the aggregation.